### ISSUE TITLE
NullPointerException when last line of input is empty or incomplete

### ISSUE THREAD

**Issue by the-gobleo-tinkerer** [ZbychuButItWasTaken/NOAA_HIRS_Decoder#2]
If the input minorFrames file has an empty or incomplete line at the end, after the "Reading data" progress bar gets to 100%, this error is thrown:

```
Loading file....
Newest file found: minorFrames_20201225_142801.txt
OK!

Converting data....
[==========101%==========]
OK!

Readnig data....
[==========100%==========]Program encountered an unhandled Exception:
java.lang.NullPointerException
	at sh.surge.NOAA_HIRS_Decoder.Main.decode(Main.java:967)
	at sh.surge.NOAA_HIRS_Decoder.Main.main(Main.java:180)
```

For example, a file like the following will cause the error to be thrown (note the empty line at the end):

```
...
371.65254 ED E2 08 89 22 43 7C 00 3F DA 00 00 12 F5 C2 C2 CE 02 C0 00 3F BF BB E8 A0 00 03 6E 07 FF EA 1D 00 00 08 D0 E0 00 01 CE 01 00 08 BB 80 04 0F FE 8C 00 0F FF FF FF F8 40 88 00 01 9C B8 00 00 E4 40 00 05 E0 B8 00 02 12 C0 00 06 97 50 00 84 1B 20 01 05 20 21 39 E1 80 06 49 30 00 00 3D 70 00 06 A0 6F D8 80 72 71 1C 
371.85256 ED E2 08 21 A2 65 08 20 6A 01 00 69 0A F4 50 98 4E E6 2F E2 E8 7C C9 22 27 5F FA B5 A1 FD 58 E4 B3 EC 06 67 00 01 0E 10 0E 8C 62 5C 71 C6 81 FE 7C FF 18 01 19 BC D8 03 9A 6F 7C 65 88 26 47 46 1B 5D EF 65 C0 00 14 96 40 00 14 BA 80 00 30 5F 00 80 08 D9 47 C7 80 00 1A 49 40 00 21 EF C0 00 06 AB 4E 93 C2 B6 95 4E 

```

The following will also cause the same error (incomplete line at the end):

```
...
371.65254 ED E2 08 89 22 43 7C 00 3F DA 00 00 12 F5 C2 C2 CE 02 C0 00 3F BF BB E8 A0 00 03 6E 07 FF EA 1D 00 00 08 D0 E0 00 01 CE 01 00 08 BB 80 04 0F FE 8C 00 0F FF FF FF F8 40 88 00 01 9C B8 00 00 E4 40 00 05 E0 B8 00 02 12 C0 00 06 97 50 00 84 1B 20 01 05 20 21 39 E1 80 06 49 30 00 00 3D 70 00 06 A0 6F D8 80 72 71 1C 
371.85256 ED E2 08 21 A2 65 08 20 6A 01 00
```

**Comment by sp5ews** [comment]
Hmmm I had a simmilar issue one time, I thought I fixed it. I'll have a look tomorrow. Meanwhile, happy new year!

**Comment by the-gobleo-tinkerer** [comment]
For your reference, I was using the 1.0.3 version jar downloaded from the surge.sh website. I haven't figured out how to build from source so I haven't tested the code in master. (How about putting build instructions in README?)

I'm running the jar file directly with `java -jar ...` because I'm on Linux. I'll try on Windows with the exe wrapper and report back.

Happy New Year to you too!

**Comment by the-gobleo-tinkerer** [comment]
Slightly strange, but on Windows one newline at the end won't cause the error, only two or more newlines will. Incomplete lines still cause the error.

I noticed that when opened under Linux, my test file has no newline at the end, but when opened in Windows, Notepad does show a newline at the end of the file. 

I think this might be because Windows and Linux represent newlines differently.

**Comment by sp5ews** [comment]
It should be fixed now, in v1.0.3.1. I will include the instructions on compiliing in the future, shouldn't be too hard
